core/runtime/runtime.go

```diff
@@ -12,3 +12,9 @@ import (
 func WithCaptureHTTPResponse(parent context.Context, resp **http.Response) context.Context {
 	return context.WithValue(parent, config.ContextHTTPResponse, resp)
 }
+
+// WithCaptureHTTPRequest adds the raw HTTP request to the parent context.
+// The req parameter will contain the raw HTTP request.
```

Comment:
`HTTP request retrieval annotation` because `WithCaptureHTTPRequest` doesn't add the raw request to the context by itself, it just adds that "annotation" to the context to signal that it should be captured when the request is executed